### ISSUE TITLE
[Bug]:When paste and undo are used together, the calculated value is wrong

### ISSUE THREAD

**Issue by nomind2think** [handsontable/handsontable#11290]
### Description

1. paste 1 to A1(or range A1:AX) twice
2. undo(ctrl+z) 
3. formula sum value is 1,but A1 null value

### Video or screenshots

![image](https://github.com/user-attachments/assets/2415fd0a-cbdb-4bfc-b882-d2c0950e6ced)

https://github.com/user-attachments/assets/4cd4a6ac-2bbb-43ff-bda5-eb3449681a8b

### Demo

https://stackblitz.com/edit/react-h89p6p?file=src%2Findex.js

### HyperFormula version

2.7.0

### Your framework

vue3

### Your environment

nodejs v18.17.1

**Comment by nomind2think** [comment]
> Hi @nomind2think
> 
> Provided written steps do not align with what is shown in the video. I tried to copy a value, sum it, and undo but it all works well.
> 
> Could you please work on a less complex demo? Maybe that would help. And provide your OS/Browser settings.
> 
> ps. I transferred the issue to the Handsontable repository

Yes, I mean paste (ctrl+v) twice, then undo once (ctrl+z), then a1 will be blank, but the calculation result is 1.

**Comment by nomind2think** [comment]
@AMBudnik 
https://github.com/user-attachments/assets/3277a9a4-1e59-4b88-ba1d-65f6989b15a6

1. input 1
2. copy 1
3. paste 1
4. paste 1
5. undo

**Comment by nomind2think** [comment]
https://github.com/user-attachments/assets/ca349a3e-64c8-4f88-82d6-abb0e392703c

**Comment by AMBudnik** [comment]
Thank you for sharing the steps. Now I was able to replicate the issue.

I will report it internally and update you as soon as we fix it.

**Comment by nomind2think** [comment]
ok,thank you~

**Comment by adrianspdev** [comment]
Hi @nomind2think,

This was fixed in #12056 ("Fix undo/redo stack desync with formulas engine"), shipped in 17.0.1. The root cause was the undo/redo stack getting out of sync with the formulas engine when consecutive changes produced no net value diff.

Verified on the post-fix build with three scenarios — all behave correctly:

1. populateFromArray(0, 0, [[1]]) twice on A1 with B1 = =SUM(A1), then undo twice → A1 and B1 both restore.
2. Real CopyPaste plugin paste of 1 to A1 twice, then undo → consistent.
3. Ranged paste of 1 over A1:A5 twice with B1 = =SUM(A1:A5), then undo → SUM matches column data.

On 16.2.0 (last pre-fix release) the original symptom still reproduces, which confirms the diagnosis.

Closing this as resolved. If you upgrade to 17.0.1 (or newer) and still see the problem, please reopen this issue with an updated reproduction, or open a new one — we'll take another look.
